common/src/main/kotlin/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketRepositoryImpl.kt

```diff
@@ -47,98 +30,40 @@ import io.homeassistant.companion.android.common.data.websocket.impl.entities.Te
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.suspendCancellableCoroutine
-import kotlinx.coroutines.sync.Mutex
-import kotlinx.coroutines.sync.withLock
-import kotlinx.coroutines.withTimeoutOrNull
-import okhttp3.OkHttpClient
-import okhttp3.Request
-import okhttp3.Response
-import okhttp3.WebSocket
 import okhttp3.WebSocketListener
-import okio.ByteString
-import okio.ByteString.Companion.toByteString
-import timber.log.Timber
 
-class WebSocketRepositoryImpl @AssistedInject constructor(
-    private val okHttpClient: OkHttpClient,
+private val matterTimeout = 2.minutes
+
+class WebSocketRepositoryImpl internal constructor(
+    private val webSocketCore: WebSocketCore,
     private val serverManager: ServerManager,
-    @Assisted private val serverId: Int
 ) : WebSocketRepository, WebSocketListener() {
 
-    companion object {
-        private const val SUBSCRIBE_TYPE_ASSIST_PIPELINE_RUN = "assist_pipeline/run"
-        private const val SUBSCRIBE_TYPE_SUBSCRIBE_EVENTS = "subscribe_events"
-        private const val SUBSCRIBE_TYPE_SUBSCRIBE_ENTITIES = "subscribe_entities"
-        private const val SUBSCRIBE_TYPE_SUBSCRIBE_TRIGGER = "subscribe_trigger"
-        private const val SUBSCRIBE_TYPE_RENDER_TEMPLATE = "render_template"
-        private const val SUBSCRIBE_TYPE_PUSH_NOTIFICATION_CHANNEL =
-            "mobile_app/push_notification_channel"
-        private const val EVENT_STATE_CHANGED = "state_changed"
-        private const val EVENT_AREA_REGISTRY_UPDATED = "area_registry_updated"
-        private const val EVENT_DEVICE_REGISTRY_UPDATED = "device_registry_updated"
-        private const val EVENT_ENTITY_REGISTRY_UPDATED = "entity_registry_updated"
-
-        private const val DISCONNECT_DELAY = 10000L
+    override fun getConnectionState(): WebSocketState? {
+        return webSocketCore.getConnectionState()
     }
 
-    private val ioScope = CoroutineScope(Dispatchers.IO + Job())
-    private val mapper = jacksonObjectMapperForHACore()
-    private val activeMessages = Collections.synchronizedMap(mutableMapOf<Long, WebSocketRequest>())
-    private val id = AtomicLong(1)
-    private var connection: WebSocket? = null
-    private var connectionState: WebSocketState? = null
-    private var connectionHaVersion: HomeAssistantVersion? = null
-    private val connectedMutex = Mutex()
-    private var connected = CompletableDeferred<Boolean>()
-    private val eventSubscriptionMutex = Mutex()
-
-    private val server get() = serverManager.getServer(serverId)
-
-    private val messageQueue = Channel<Job>(capacity = Channel.UNLIMITED).apply {
-        ioScope.launch {
-            consumeEach { it.join() } // Run a job, and wait for it to complete before starting the next one
-        }
+    override fun shutdown() {
+        return webSocketCore.shutdown()
     }
 
-    override fun getConnectionState(): WebSocketState? = connectionState
-
+    // All of this should be moved and injected through a delegate
```

Comment:
Is this a todo comment?

Reply:
TODO, my idea is to have domain specific interface for instance Matter so when injected my Dagger we can get an interface for Matter instead of the whole WS impl. But it is not a must have I would say.

My feeling is that over time this class can grow significantly and it might becomes hard to use with too many methods. That's just a feeling and we can keep it like this for now.

Also the idea is that by using delegation we could keep the existing code as is without changing anything.

Reply:
What would be the benefit of multiple interfaces for the Home Assistant app, if it is the same implementation? It is not an external API with strict requirements.

As you're already saying it's not a must have, I would suggest removing the comment as there is no final decision yet.